internal/printer/printer_test.go

```diff
@@ -283,20 +297,33 @@ const someProps = {
 				Scope:       "astro-XXXX",
 				InternalURL: "http://localhost:3000/",
 			})
-			output := string(result.Output)
+			output := strings.TrimSpace(test_utils.Dedent(string(result.Output)))
 
 			toMatch := INTERNAL_IMPORTS
 			toMatch = toMatch + fmt.Sprintf("%s%s", tt.want.imports, PRELUDE)
 			if tt.want.frontmatter != "" {
+				// format want
 				toMatch = toMatch + fmt.Sprintf(`
 // ---
-
 %s
 // ---
 `, strings.TrimSpace(tt.want.frontmatter))
+				// format output
+				formatted := make([]string, 0)
+				for _, part := range strings.Split(output, "// ---") {
+					formatted = append(formatted, strings.TrimSpace(part))
+				}
+				output = strings.Join(formatted, "\n// ---\n")
```

Comment:
Since the diff library we’re using is very cranky, I put a little more work into normalizing the whitespace within the frontmatter.

Maybe we should do this in the compiler itself?

Reply:
Yeah I'd be +1 on doing this in the compiler itself!